scalding-core/src/main/scala/com/twitter/scalding/typed/TypedPipe.scala

```diff
@@ -590,10 +590,9 @@ sealed trait TypedPipe[+T] extends Serializable {
    * @return a pipe equivalent to the current pipe.
    */
   def write(dest: TypedSink[T])(implicit flowDef: FlowDef, mode: Mode): TypedPipe[T] = {
-    // Make sure that we don't render the whole pipeline twice:
-    val res = fork
-    dest.writeFrom(res.toPipe[T](dest.sinkFields)(flowDef, mode, dest.setter))
-    res
+    dest.writeFrom(toPipe[T](dest.sinkFields)(flowDef, mode, dest.setter))
+    // We want to fork after this point
+    fork
```

Comment:
why did we make this change? I guess it should be ok now with the better optimizer? i'm guessing before we needed to do it the first way to avoid re-computation?

Reply:
I'm not sure why it was ever the other way. Many `.writes` are not consumed. But by putting the fork before, you are always going to do some extra boxing in and out of a single tuple.

I can revert it, I don't think tests pass or fail either way. We really need a better set of performance vs just correctness tests.